operator/standalone/create.go

```diff
@@ -0,0 +1,236 @@
+	return pipeline.NewPipeline().
+		WithSteps(
+			pipeline.NewStepFromFunc("fetch operator config", p.FetchOperatorConfig),
+
+			pipeline.NewPipeline().WithNestedSteps("compile helm values",
+				pipeline.NewStepFromFunc("read template values", p.UseTemplateValues),
+				pipeline.NewStepFromFunc("override template values", p.OverrideTemplateValues),
+				pipeline.NewStepFromFunc("apply values from instance", p.ApplyValuesFromInstance),
+			),
+
+			pipeline.NewPipeline().WithNestedSteps("deploy resources",
+				pipeline.NewStepFromFunc("ensure deployment namespace", p.EnsureDeploymentNamespace),
+				pipeline.NewStepFromFunc("ensure credentials secret", p.EnsureCredentialsSecret),
+				pipeline.NewStepFromFunc("ensure helmrelease exists", p.EnsureHelmRelease),
+			),
+		).
+		RunWithContext(ctx).Err()
+}
+
+// FetchOperatorConfig fetches a matching v1alpha1.PostgresqlStandaloneOperatorConfig from the OperatorNamespace.
+// The Major version specified in v1alpha1.PostgresqlStandalone is used to filter the correct config by the v1alpha1.PostgresqlMajorVersionLabelKey label.
+// If there is none or multiple found, it returns an error.
+func (p *CreateStandalonePipeline) FetchOperatorConfig(ctx context.Context) error {
+	list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
+	labels := client.MatchingLabels{
+		v1alpha1.PostgresqlMajorVersionLabelKey: p.instance.Spec.Parameters.MajorVersion.String(),
+	}
+	ns := client.InNamespace(p.operatorNamespace)
+	err := p.client.List(ctx, list, labels, ns)
+	if err != nil {
+		return err
+	}
+	if len(list.Items) == 0 {
+		return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	if len(list.Items) > 1 {
+		return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	p.config = &list.Items[0]
+	return nil
+}
+
+// UseTemplateValues copies the Helm values and Chart metadata from the v1alpha1.PostgresqlStandaloneOperatorConfig spec as the starting parameters.
+//
+// This step assumes that the config has been fetched first via FetchOperatorConfig.
+func (p *CreateStandalonePipeline) UseTemplateValues(_ context.Context) error {
+	values := HelmValues{}
+	err := values.Unmarshal(p.config.Spec.HelmReleaseTemplate.Values)
+	p.helmValues = values
+	p.helmChart = &p.config.Spec.HelmReleaseTemplate.Chart
+	return err
+}
+
+// OverrideTemplateValues searches for a specific HelmRelease spec that matches the Chart version from the template spec.
+// If it does, the template values are replaced or merged.
+//
+// This step assumes that the config has been fetched first via FetchOperatorConfig.
+func (p *CreateStandalonePipeline) OverrideTemplateValues(_ context.Context) error {
+	for _, release := range p.config.Spec.HelmReleases {
+		// TODO: maybe a better semver comparison later on?
```

Comment:
IMHO only if we want to support downgrades?

Reply:
I thought more in the direction of something like ">= 11.0", e.g. if a helm release override works for all charts beginning from 11.0 onwards (and not just a very specific chart version). Right now, one needs to provide overrides for every single chart version that requires overridden values. Could be cumbersome to manage, though somewhat easier using jsonnet/Commodore :shrug:

Reply:
Ah I see the issue, yes makes sense.